velox/exec/PrefixSort.cpp

```diff
@@ -133,12 +147,13 @@ PrefixSortLayout PrefixSortLayout::makeSortLayout(
       break;
```

Comment:
We need do this check again after we looping over all keys.  We need to be extra careful about a super long string value makes the normalized key very long and kill the performance; in that case we should drop out of prefix sort and fallback.

Reply:
I moved the check after getting the encodeSize. I also added test to make sure normalized keys would not exceed the threshold. Thanks.